chall09/isoria-g.c

```diff
@@ -0,0 +1,335 @@
+{
+    unsigned long long      c;
+
+    if (n < 0)
+        c = ft_size(ft_value(n)) + 2;
+    else
+        c = ft_size(ft_value(n)) + 1;
+    return (c);
+}
+
+static char *ft_zero(char *p)
+{
+    char    *pp;
+
+    pp = (char *)malloc(1 * sizeof(char) + 1);
+    pp = p;
+    pp[0] = '0';
+    pp[1] = '\0';
+    return (pp);
+}
+
+char                    *ft_itoa(unsigned long long n)
+{
+    unsigned long long  a;
+    unsigned long long  b;
+    char                *p;
+
+    a = 0;
+    b = ft_value(n);
+    if (!(p = (char *)malloc((ft_size(b) + 2) * sizeof(char) + 1)))
+        return (NULL);
+    if (n == 0)
+        return (ft_zero(p));
+    if (n < 0)
+        p[a] = 45;
+    if (n < 0)
+        a = ft_size(ft_value(n)) + 1;
+    else
+        a = ft_size(ft_value(n));
+    while (b > 9)
+    {
+        p[a--] = (b % 10) + 48;
+        b = b / 10;
+    }
+    p[a] = (b % 10) + 48;
+    p[ft_end(n)] = '\0';
+    return (p);
+}
+
+char    *ft_isspace(char *str)
+{
+    int i;
+
+    i = 0;
+    while ((str[i] >= 9 && str[i] <= 13) || str[i] == 32)
+        i++;
+    return (&str[i]);
+}
+
+unsigned long long      ft_atoi(char *str)
+{   
+    char                *ptr;
+    int                 i;
+    int                 j;
+    int                 k;
+    unsigned long long  nb;
+    unsigned long long  total;
+
+    i = 0;
+    k = 0;
+    ptr = ft_isspace(str);
+    if (ptr[0] < '0' || ptr[0] > '9')
+        return (-1);
+    j = 0;
+    while ((ptr[j] >= '0' && ptr[j] <= '9') || ptr[j] == 32)
+    {
+        if (ptr[j] == 32)
+            k++;
+        if ((ptr[j] >= '0' && ptr[j] <= '9') && k > 0)
+            return(-1);
+        j++;
+    }
+    if (ptr[j] != '\0')
+        return (-1);
+    i = 0;
+    total = 0;
+    while (i < j - k)
+    {
+        nb = ptr[i] - '0';
+        total = total + nb;
+        if (i < j - k - 1)
+            total = total * 10;
+        i++;
+    }
+    return (total);
+}
+
+char                    *ft_format_duration(char *seconds)
+{
+    unsigned long long  nb;
+    unsigned long long  yy;
+    unsigned long long  dd;
+    unsigned long long  hh;
+    unsigned long long  mm;
+    unsigned long long  ss;
+    char *str = "";
+
+    if (seconds == NULL)
+        return("\"Invalid input.\"");
+    nb = ft_atoi(seconds);
+    if (nb == -1)
+        return("\"Invalid input.\"");
+    if (nb == 0)
+        return("\"now\"");
+    yy = nb / 31536000;
+    dd = (nb % 31536000) / 86400;
+    hh = ((nb % 31536000) % 86400) / 3600;
+    mm = (((nb % 31536000) % 86400) % 3600) / 60;
+    ss = (((nb % 31536000) % 86400) % 3600) % 60;
+    str = ft_strjoin(str, "\"");
+    if (yy > 0)
+    {
+        if (yy == 1)
+        {
+            str = ft_strjoin(str, ft_itoa(yy));
+            str = ft_strjoin(str, " year");
+        }
+        else
+        {
+            str = ft_strjoin(str, ft_itoa(yy));
+            str = ft_strjoin(str, " years");
+        }
```

Comment:
This seems overly repetitive. You should have made a function for this.  
Even better you could have used functions like `sprintf` or other variations of it to format your string easily